api-reference/10 UI Components/dxTreeList/1 Configuration/toolbar.md

```diff
@@ -6,7 +6,61 @@ inheritsType: dxTreeListToolbar
+  </tr>
+  <tr>
+    <td><img src="/images/DataGrid/icons/columnChooser.png" alt="DevExtreme HTML5 JavaScript jQuery Angular Knockout UI component TreeList Toolbar ColumnChooserButton" style="margin:1px; vertical-align:middle"/></td>
+    <td><i>columnChooserButton</i></td>
+    <td>columnChooser.<a href="/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columnChooser/#enabled">enabled</a> is set to <b>true</b></td>
+  </tr>
+  <tr>
+    <td><img src="/images/DataGrid/icons/toolbar_revert.png" alt="DevExtreme HTML5 JavaScript jQuery Angular Knockout UI component TreeList Toolbar RevertButton" style="margin:1px; vertical-align:middle"/></td>
+    <td><i>revertButton</i></td>
+    <td>editing.<a href="/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode">mode</a> is set to <i>"batch"</i></td>
+  </tr>
+  <tr>
+    <td><img src="/images/DataGrid/icons/toolbar_save.png" alt="DevExtreme HTML5 JavaScript jQuery Angular Knockout UI component TreeList Toolbar SaveButton" style="margin:1px; vertical-align:middle"/></td>
+    <td><i>saveButton</i></td>
+    <td>editing.<a href="/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#mode">mode</a> is set to <i>"batch"</i> and editing.<a href="/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/editing/#allowUpdating">allowUpdating</a> is set to <b>true</b></td>
+  </tr>
+  <tr>
+    <td><img src="/images/DataGrid/icons/toolbar_searchPanel.png" alt="DevExtreme HTML5 JavaScript jQuery Angular Knockout UI component TreeList Toolbar searchPanel" style="margin:1px; vertical-align:middle"/></td>
+    <td><i>searchPanel</i></td>
+    <td>searchPanel.<a href="/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/searchPanel/#visible">visible</a> is set to <b>true</b></td>
+  </tr>
+</table>
+
+To add a predefined item, add its string to the toolbar array. If you need to configure an item, add an object instead and specify the [name](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/toolbar/items/#name) and other [properties](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/toolbar/items/).
+
+**Custom Items**
+Custom items can display any html code.  To add such an item, specify its [template](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/toolbar/items/#template) and other properties. To embed a DevExtreme component, specify the [widget](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/toolbar/items/#widget) and other [properties](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/toolbar/items/) instead.
+
+[note] If you use toolbar to specify a custom item and/or configure a predefined item, add corresponding strings for other used features. Otherwise, the TreeList will not display these controls.
```

Comment:
Make the descriptions based on the [Toolbar Customization](https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/ToolbarCustomization/jQuery/Light/) demo.

```suggestion
- DevExtreme components
...

- Custom elements
...
```